catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/plugins/FindChildrenStreamEndPlugin.java

```diff
@@ -0,0 +1,192 @@
+
+        context.modifyParentOrChild(new Function<AtomicBoolean, Void>() {
+            @Override
+            public Void apply(AtomicBoolean isParentDirty) {
+
+                if (isParentDirty.get()) {
+                    context.getParentMetacard()
+                            .ifPresent(parentMetacard -> handleParentMetacard(context,
+                                    parentMetacard));
+
+                    isParentDirty.set(false);
+                }
+
+                return null;
+            }
+        });
+
+    }
+
+    private void handleParentMetacard(Context context, Metacard parentMetacard) {
+
+        Handler handler = factory.build();
+
+        Filter filter = filterBuilder.attribute(Associations.DERIVED)
+                .is()
+                .equalTo()
+                .text(parentMetacard.getId());
+
+        int startIndex = 1;
+        Long expectedReturnCount = null;
+        int subsequentErrorCount = 0;
+
+        do {
+            QueryRequest queryRequest = new QueryRequestImpl(new QueryImpl(filter,
+                    startIndex,
+                    BATCH_SIZE,
+                    new SortByImpl(Core.METACARD_CREATED, SortOrder.ASCENDING),
+                    true,
+                    0));
+
+            try {
+                QueryResponse queryResponse = context.getUdpStreamProcessor()
+                        .getCatalogFramework()
+                        .query(queryRequest);
+
+                if (startIndex == 1) {
+                    expectedReturnCount = queryResponse.getHits();
+                }
+
+                List<Metacard> batch = queryResponse.getResults()
+                        .stream()
+                        .map(Result::getMetacard)
+                        .collect(Collectors.toList());
+
+                handler.handle(context, parentMetacard, batch);
+
+                startIndex += batch.size();
+
+                subsequentErrorCount = 0;
+
+            } catch (UnsupportedQueryException | SourceUnavailableException | FederationException e) {
+                LOGGER.debug("unable to find the children for a parent metacard: metacardId={}",
+                        parentMetacard.getId(),
+                        e);
+                subsequentErrorCount++;
+            }
+
+        } while (continueProcessing(startIndex, expectedReturnCount, subsequentErrorCount));
+
+        handler.end(context, parentMetacard);
+
+    }
+
+    private boolean continueProcessing(int startIndex, Long expectedReturnCount,
+            int subsequentErrorCount) {
+        return isSubsequentErrorCountBelowLimit(subsequentErrorCount) && (
+                isExpectedReturnCountUninitialized(expectedReturnCount) || areBatchesRemaining(
+                        startIndex,
+                        expectedReturnCount));
+    }
+
+    private boolean areBatchesRemaining(int startIndex, long expectedReturnCount) {
+        return startIndex <= expectedReturnCount;
```

Comment:
added null check, but should never actually happen because of short-circuit logic.